flow/Makefile

```diff
@@ -332,11 +334,13 @@ synth: versions.txt \
 #-------------------------------------------------------------------------------
 SYNTH_SCRIPT ?= $(SCRIPTS_DIR)/synth.tcl
 
-$(RESULTS_DIR)/1_1_yosys.v: $(DONT_USE_LIBS) $(WRAPPED_LIBS) $(DONT_USE_SC_LIB) $(DFF_LIB_FILE) $(VERILOG_FILES) $(CACHED_NETLIST) $(LATCH_MAP_FILE) $(ADDER_MAP_FILE)
-	mkdir -p $(RESULTS_DIR) $(LOG_DIR) $(REPORTS_DIR)
+$(SYNTH_HIER_REPORT):
 ifeq ($(SYNTH_HIERARCHICAL), 1)
 	($(TIME_CMD) $(YOSYS_CMD) $(YOSYS_FLAGS) -c $(HIER_REPORT_SCRIPT)) 2>&1 | tee $(LOG_DIR)/1_1_yosys_hier_report.log
 endif
```

Comment:
no need to wrap with `if` here. I think it would be better to have this be a normal target and just to modify the dependencies for `1_1_yosys.v`
```
ifeq ($(SYNTH_HIERARCHICAL), 1)
$(RESULTS_DIR)/1_1_yosys.v: $(OBJECTS_DIR)/mark_hier_stop_modules.tcl
endif
```
This will add the dependency when it's needed/requested, but otherwise will not cause this step to execute.